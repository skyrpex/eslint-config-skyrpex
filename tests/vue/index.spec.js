const { CLIEngine } = require("eslint");
const { cleanReport } = require("../utils");

const cwd = `${__dirname}/fixtures`;

test("should validate without errors", () => {
    const engine = new CLIEngine({
        cwd,
        configFile: require.resolve("./eslint.config.js"),
        useEslintrc: false,
        ignore: false,
    });
    const report = engine.executeOnFiles(["component.fixture.vue"]);
    expect(cleanReport(report)).toMatchSnapshot();
});

test("should report buble-related errors", () => {
    const engine = new CLIEngine({
        cwd,
        configFile: require.resolve("./eslint-buble.config.js"),
        useEslintrc: false,
        ignore: false,
    });
    const report = engine.executeOnFiles(["component.fixture.vue"]);
    expect(cleanReport(report)).toMatchSnapshot();
});
