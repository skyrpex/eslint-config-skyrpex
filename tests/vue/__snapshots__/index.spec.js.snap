// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should report buble-related errors 1`] = `
Object {
  "errorCount": 1,
  "fixableErrorCount": 1,
  "fixableWarningCount": 0,
  "results": Array [
    Object {
      "errorCount": 1,
      "filePath": "tests/vue/fixtures/component.fixture.vue",
      "fixableErrorCount": 1,
      "fixableWarningCount": 0,
      "messages": Array [
        Object {
          "column": 27,
          "endColumn": 28,
          "endLine": 16,
          "fix": Object {
            "range": Array [
              285,
              286,
            ],
            "text": "",
          },
          "line": 16,
          "message": "Delete \`,\`",
          "nodeType": null,
          "ruleId": "prettier/prettier",
          "severity": 2,
          "source": "    testVeryLongMethodName,",
        },
      ],
      "source": "<template>
    <div>
        <h1
            class=\\"heading\\"
            :class=\\"cssClasses\\"
        />
    </div>
</template>

<script>
const testVeryLongMethodName = () => {};

testVeryLongMethodName(
    testVeryLongMethodName,
    testVeryLongMethodName,
    testVeryLongMethodName,
);

export default {
    data: () => ({
        cssClasses: \\"heading--dark\\",
    }),
};
</script>
",
      "warningCount": 0,
    },
  ],
  "warningCount": 0,
}
`;

exports[`should validate without errors 1`] = `
Object {
  "errorCount": 0,
  "fixableErrorCount": 0,
  "fixableWarningCount": 0,
  "results": Array [
    Object {
      "errorCount": 0,
      "filePath": "tests/vue/fixtures/component.fixture.vue",
      "fixableErrorCount": 0,
      "fixableWarningCount": 0,
      "messages": Array [],
      "warningCount": 0,
    },
  ],
  "warningCount": 0,
}
`;
